docs/src/references/agent/device-management-api.md

```diff
@@ -0,0 +1,325 @@
+
+### Command metadata topics
+
+The __command metadata topics__ are used to declare which commands are available for a device,
+and, if so, to which extent.
+
+```mermaid
+graph LR
+  te --/--- identifier --/--- cmd
+  subgraph root
+    te
+  end
+
+  subgraph target
+    identifier["&lt;target identifier&gt;"]
+  end
+
+  subgraph command["command metadata"]
+    direction LR
+    cmd --/--- cmd_type["&lt;command_type&gt;"]
+  end
+```
+
+Where the groups are described as follows:
+
+| Group        | Description                                                                                                                                                                              |
+|--------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| root         | The [topic root prefix](../mqtt-api.md/#group-root) for this installation of thin-edge.                                                                                                  |
+| target       | The [topic identifier](../mqtt-api.md/#group-identifier) of the target device, possibly a [custom identifier](/contribute/design/mqtt-topic-design.md/#using-custom-identifier-schemas). |
+| `cmd`        | The [command channel](../mqtt-api.md/#channel-identifier) grouping of all the commands for this target.                                                                                  |
+| command_type | The type name of the operation.                                                                                                                                                          |
+
+A service that implements an operation for a device publishes on start a message notifying on the MQTT Bus
+that this device can be sent commands of this type.
+
+These messages are published with the retained flag set. So, a client process, such a mapper, can discover on start
+what are __all the capabilities of all the devices__:
+
+```sh te2mqtt formats="v1"
+tedge mqtt sub 'te/+/+/+/+/cmd/+'
+```
+
+### Command status topics
+
+The actual command requests are published on the __command status topics__.
+For each request, a specific command topic is created to monitor the progress of the command from its initial state to its completion.
+These topics are named using a unique command identifier forged by the requester.
+
+```mermaid
+graph LR
+  te --/--- identifier --/--- cmd
+  subgraph root
+    te
+  end
+
+  subgraph target
+    identifier["&lt;target identifier&gt;"]
+  end
+
+  subgraph command["command request"]
+    direction LR
+    cmd --/--- cmd_type["&lt;command_type&gt;"] --/--- cmd_id["&lt;cmd_id&gt;"]
+  end
+```
+
+| Group        | Description                               |
+|--------------|-------------------------------------------|
+| command_id   | The identifier of the operation instance. |
```

Comment:
It might be worth mentioning that the `command_id` is an arbitrary string however it should be unique. It is recommended to either use a unique id generator, or add a unix timestamp as a suffix, e.g. `date +%s`

Reply:
Added